### ISSUE TITLE
Mounting routers defined programatically

### ISSUE THREAD

**Issue by jlfsjunior** [rstudio/plumber#709]
First of all, thanks a lot for v1.0 and the improvements made in the documentation.

My silly issue is related to the mounting procedure described [here](https://www.rplumber.io/articles/programmatic-usage.html#mount-static): I can't make it work if the nested router is also defined programatically. What I mean by that is the following:

Given `test.R`
```R
# test.R
#* @get /
function() {
  "Test annotations"
}
````

and `test_pr.R`
```R
# test_pr.R
pr() %>% 
  pr_get(
    path = "/", 
    handler = function(req, res){
      "Test pr"
    })
```

I would expect both routers to be mounted with a single endpoint each:
```R
test <- pr("test.R")
test_pr <- pr("test_pr.R")

root <- pr() %>% 
  pr_mount("/test", test_pr) %>% 
  pr_mount("/testpr", test_pr)
```

However, `/testpr` mounts with no endpoints:
```R
> root
# Plumber router with 0 endpoints, 4 filters, and 2 sub-routers.
# Call run() on this object to start the API.
├──[queryString]
├──[postBody]
├──[cookieParser]
├──[sharedSecret]
├──/test
│  │ # Plumber router with 1 endpoint, 4 filters, and 0 sub-routers.
│  ├──[queryString]
│  ├──[postBody]
│  ├──[cookieParser]
│  ├──[sharedSecret]
│  ├──/ (GET)
├──/testpr
│  │ # Plumber router with 0 endpoints, 4 filters, and 0 sub-routers.
│  ├──[queryString]
│  ├──[postBody]
│  ├──[cookieParser]
│  ├──[sharedSecret]
```

In other words, how should `users.R` (or `products.R`) look like in its programmatic version?

**Comment by jlfsjunior** [comment]
Ok, I think I can live with the way done in [this example](https://github.com/rstudio/plumber/blob/master/inst/plumber/12-entrypoint/using_plumber_tag.R).

**Comment by schloerke** [comment]
@jlfsjunior We need to make this clearer in the docs, but I would set up the codes like...


Regular `plumb()`-able file
```r
# test.R

#* @get /
function() {
  "Test annotations"
}
```

Programmatically created router inside a `@plumber` tag. Allows for the file to be `plumb()`ed.
```r
# test_pr.R

#' @plumber
function(pr) {
  pr %>% 
    pr_get(
      path = "/", 
      handler = function(req, res){
        "Test pr"
      }
    )
}
```

Main plumber file `plumb()`s files using `pr()` and mounts them appropriately.  This file (and any of the other files) can include regular routes / routine `plumb()`-able tags.

```r
# plumber.R

#* @plumber
function(pr) {
  test <- pr("test.R")
  test_pr <- pr("test_pr.R")
  
  pr %>% 
    pr_mount("/test", test_pr) %>% 
    pr_mount("/testpr", test_pr)
}
```


This also allows for you to call `plumb(dir = ".")`, which will default to `plumber.R`.
